component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/moesif/ParallelQueueWorker.java

```diff
@@ -20,45 +20,94 @@
-import org.wso2.am.analytics.publisher.retriever.MoesifKeyRetriever;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.BlockingQueue;
 
 /**
- * Will dequeue the events from queues and send then to the moesif client {@link MoesifClient}.
+ * Will dequeue the events from queues and send then to the moesif client.
  */
 public class ParallelQueueWorker implements Runnable {
     private static final Logger log = LogManager.getLogger(ParallelQueueWorker.class);
     private BlockingQueue<MetricEventBuilder> eventQueue;
-    private MoesifKeyRetriever keyRetriever;
-    private MoesifClient client;
+    private AbstractMoesifClient client;
+    private final int batchSize = 50;
+    private final long batchTimeoutMs = 2000;
 
-    public ParallelQueueWorker(BlockingQueue<MetricEventBuilder> queue, MoesifClient moesifClient) {
+    public ParallelQueueWorker(BlockingQueue<MetricEventBuilder> queue, AbstractMoesifClient moesifClient) {
         this.eventQueue = queue;
         this.client = moesifClient;
     }
 
-    public void run() {
+    /**
+     * Continuously runs a worker thread that processes analytics events in batches from a blocking queue.
+     * <p>
+     * The method polls the {@code eventQueue} at fixed intervals (100ms) for new {@code MetricEventBuilder} events.
+     * Events are collected into a batch and processed when either:
+     * <ul>
+     *   <li>The batch reaches the configured {@code batchSize}</li>
+     *   <li>The elapsed time since the last batch processing exceeds {@code batchTimeoutMs}</li>
+     * </ul>
+     * Once either condition is met, the collected batch is sent using {@code processBatch}, and the timer is reset.
+     * <p>
+     * If the thread is interrupted, it exits gracefully by setting the interrupt flag. Any other exception
+     * during processing is logged, and the affected events will be dropped.
+     *
+     * This method is intended to be executed in a dedicated thread via an {@code ExecutorService} or similar mechanism.
+     */
 
+    public void run() {
+        List<MetricEventBuilder> batch = new ArrayList<>(batchSize);
+        long lastBatchTime = System.currentTimeMillis();
         while (true) {
             MetricEventBuilder eventBuilder;
             try {
-                eventBuilder = eventQueue.take();
+
+                eventBuilder = eventQueue.poll(100, java.util.concurrent.TimeUnit.MILLISECONDS);
                 if (eventBuilder != null) {
-                    client.publish(eventBuilder);
+                    batch.add(eventBuilder);
+                }
+
+                long currrentBatchTime = System.currentTimeMillis();
+                boolean shouldSendBatch = batch.size() >= batchSize ||
+                        (currrentBatchTime - lastBatchTime) >= batchTimeoutMs;
+
+                if (shouldSendBatch) {
+                    log.debug("Sending batch of {} events", batch.size());
```

Comment:
Does this logs for batch size of zero as well? if yes and not intended let's add a check